doc/deployment/deploy_custom.md

```diff
@@ -11,11 +22,360 @@ This document  details the various options of the `pachctl deploy custom ...` co
+- `--pachd-memory-request`: The amount of memory that Kubernetes allocates to `pachd`. This flag accepts the SI suffixes.
+- `shards`: The maximum number of `pachd` nodes allowed in the cluster. Increasing this number from the default value of 16 might result in degraded performance.
+
+
+#### Enterprise Edition features
+
+- `--dash-image`: The Docker image for the Pachyderm Enterprise Edition dashboard.
+- `--image-pull-secret`: The name of a Kubernetes secret that Pachyderm uses to pull from a private Docker registry.
+- `--no-dashboard`: Skip the creation of a manifest for the Enterprise Edition dashboard.
+- `--registry`: The registry for Docker images.
+- `--tls`:  A string in the `"<cert path>,<key path>"` format with the signed TLS certificate that is used for encrypting `pachd` communications.
+
+
+#### Output formats
+
+- `--dry-run`: Create a manifest and send it to standard output, but do not deploy to Kubernetes.
+- `-o` or `--output`: An output format. You can choose from JSON (default) or YAML.
+
+#### Logging
+- `log-level`: The `pachd` verbosity level, from most verbose to least. You can set this parameter to `debug`, `info`, or `error`.
+- `-v` or `--verbose`: Controls the verbosity of the `pachctl` invocation.
+
+#### Complete example invocation
+Since we're deploying to a code infrastructure that works with YAML files,
+you need to add the flags for that.
+Limit your configuration to local roles only by adding the `--local-roles` flag.
+Because you need to deploy your cluster to a code infrastructure that works with YAML files,
+the output will be saved to a file that the code infrastructure can later use.
+The final deployment command includes the following flags:
+```
+pachctl deploy custom --persistent-disk aws --object-store s3 \
+    foobar 10 \
+    pachyderm-bucket  'OBSIJRBE0PP2NO4QOA27' 'tfteSlswRu7BJ86wekitnifILbZam1KYY3TG' 'minio:9000' \
+    --dynamic-etcd-nodes 10
+    --local-roles --output yaml  --dry-run > custom_deploy.yaml
+```
+
+For more information about the contents of the `custom_deploy.yaml` file,
+see [Anatomy of a Pachyderm deployment manifest](#anatomy-of-a-pachyderm-deployment-manifest).
+
+## Anatomy of a Pachyderm deployment manifest
+
+By default, when you run the `pachctl deploy ...` command with the `--dry-run` flag,
+you are generating a JSON-encoded Kubernetes manifest in one stream to standard output. 
+That manifest consists of smaller manifests,
+each of which corresponds to a particular aspect of a Pachyderm deployment.
+
+Pachyderm deploys the following sets of application components:
+- `pachd`, the main Pachyderm pod
+- `etcd`, the administrative datastore for `pachd`
+- `dash`, the web-based enterprise ui for Pachyderm Enterprise Edition
+
+In general, there are two categories of manifests in the file,
+roles-and-permissions-related and application-related. 
+
+
+## Roles and permissions manifests
+
```

Comment:
all the headings under this *Roles and permissions manifests* can be put in a table. We also need an intro text before the table. Something like:
```
This table lists all the roles and permissions sections that Pachyderm configures in the deployment manifest.
```